File: chap1/src/frontend/parser.py
# -*- coding: utf-8 -*-

# package wci.frontend;
# import wci.intermediate.ICode;
# import wci.intermediate.SymTab;
from abc import ABCMeta, abstractmethod


class Parser(metaclass=ABCMeta):
    def __init__(self, scanner):
        self.symTab = None
        self.scanner = scanner
        self.iCode = None

    @abstractmethod
    def parse(self):
        raise NotImplementedError()

    @abstractmethod
    def get_error_count(self):
        raise NotImplementedError()

    def current_token(self):
        return self.scanner.cuurent_token()

    def next_token(self):
        return self.scanner.next_token()

File: Limbus/limbus_core/Parser.fs
module Parser

open Token
open Scanner

[<AbstractClass>]
type Parser(scanner : Scanner) = 
    member this.symTab = null 
    member this.iCode = null 
    member this.scanner = scanner 
    
    abstract member parse : unit
    abstract member get_error_count : int
    
    member this.current_token() : Token = this.scanner.cuurent_token()
        
    member this.next_token() : Token = this.scanner.next_token()
    